### DISCUSSION TITLE
Xamarin.ios MagickImageCollection

### DISCUSSION THREAD

**Discussion by TestTEMV4** [dlemstra/Magick.NET#1258] [Help] [answered]
Hi,

I am having trouble using MagickImageCollection in a Xamarin.iOS project. As soon as the app reaches the 
using (MagickImageCollection images = new MagickImageCollection()) statement, it throws an exception:
"The type initializer for 'NativeMagickImageCollection' threw an exception."

I read [here](https://github.com/dlemstra/Magick.NET/issues/424) that MagickAnyCPU.CacheDirectory must be instantiated but MagickAnyCPU is not available on iOS, which makes sense.

Can you help me with that?

**Comment by dlemstra** [comment, marked as answer]
Setting that `CacheDirectory` is no longer an option and will not resolve your issue. I don't think that you can use Magick.NET on iOS because the native library doesn't seem to be working on that platform.

**Reply by TestTEMV4** [reply]
OK. Thank you for clarifying that. I appreciate it.
